### PR TITLE
fix public user subscription handling for company topics

### PR DESCRIPTION
**Pull request by petrjasek**
NHUB-365

### CODE REVIEW

File: assets/user-profile/actions.ts [hunk 1]
```diff
@@ -340,3 +333,19 @@ export function updateUserNotificationSchedules(schedule: Omit<IUser['notificati
...
 }
+
+export function setTopicSubscribers(topic: ITopic, subscribers: ITopic['subscribers']) {
+    return (dispatch: any, getState: any) => {
+        const updates = {subscribers};
+        return updateTopic(topic, updates, dispatch);
+    };
+}
+
+function updateTopic(topic: ITopic, updates: any, dispatch: any) {
```
Comment (MarkLark86) [inline assets/user-profile/actions.ts:344]: Types can be improved here to enforce correct updates are supplied, `updates: Partial<ITopic>` and return type `Promise<void>`

File: assets/user-profile/actions.ts [hunk 2]
```diff
@@ -340,3 +333,19 @@ export function updateUserNotificationSchedules(schedule: Omit<IUser['notificati
             .catch((error) => errorHandler(error, dispatch, setError(error)));
     };
 }
+
+export function setTopicSubscribers(topic: ITopic, subscribers: ITopic['subscribers']) {
+    return (dispatch: any, getState: any) => {
```
Comment (MarkLark86) [inline assets/user-profile/actions.ts:338]: Lint warning: `getState` is defined but never used

File: assets/user-profile/reducers.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import {SET_USER_COMPANY_MONITORING_LIST} from 'monitoring/actions';
 import {modalReducer} from 'reducers';
 import {GET_NAVIGATIONS, QUERY_NAVIGATIONS} from 'navigations/actions';
 import {SET_TOPICS} from '../search/actions';
+import {ITopic} from 'interfaces';
```
Comment (MarkLark86) [inline assets/user-profile/reducers.ts:24]: Lint warning: `ITopic` is defined but never used

**Review by MarkLark86 — APPROVED** [review]

File: assets/user-profile/actions.ts (hunk 1 above)
Comment (petrjasek) [reply assets/user-profile/actions.ts]: added updates type, seems like return type it gets right automatically

**Pull request merged**